File: DSA/Ayush_Mishra_590014844/Day_82_Question_3.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Node {
    int data;
    struct Node *left, *right;
} Node;

Node* createNode(int val) {
    Node* node = (Node*)malloc(sizeof(Node));
    node->data = val;
    node->left = node->right = NULL;
    return node;
}

Node* insert(Node* root, int val) {
    if (!root) return createNode(val);
    if (val < root->data) root->left = insert(root->left, val);
    else root->right = insert(root->right, val);
    return root;
}

void convertToGreaterTree(Node* root, int* sum) {
    if (!root) return;
    convertToGreaterTree(root->right, sum);
    *sum += root->data;
    root->data = *sum;
    convertToGreaterTree(root->left, sum);
}

void inorder(Node* root) {
    if (!root) return;
    inorder(root->left);
    printf("%d ", root->data);
    inorder(root->right);
}

int main() {
    int n, val;
    printf("Enter number of nodes in BST: ");
    scanf("%d", &n);
    Node* root = NULL;
    printf("Enter node values: ");
    for (int i = 0; i < n; i++) {
        scanf("%d", &val);
        root = insert(root, val);
    }
    int sum = 0;
    convertToGreaterTree(root, &sum);
    printf("Inorder traversal of Greater Tree:\n");
    inorder(root);
    return 0;
}

File: DSA/VaniAgrawal_590015846/Day_82_Question_3.py
# Convert BST to Greater Tree
class Node:
    def __init__(self, val):
        self.val = val
        self.left = None
        self.right = None

def build_tree(values):
    if not values: return None
    nodes = [None if v == "null" else Node(int(v)) for v in values]
    kids = nodes[::-1]
    root = kids.pop()
    for node in nodes:
        if node:
            if kids: node.left = kids.pop()
            if kids: node.right = kids.pop()
    return root

def convert_greater_tree(root):
    total = 0
    def reverse_inorder(node):
        nonlocal total
        if not node: return
        reverse_inorder(node.right)
        total += node.val
        node.val = total
        reverse_inorder(node.left)
    reverse_inorder(root)

def inorder(root):
    return inorder(root.left) + [root.val] + inorder(root.right) if root else []

values = input().split()
root = build_tree(values)
convert_greater_tree(root)
print(*inorder(root))
